contracts/ibc-forwarder/src/contract.rs

```diff
@@ -87,9 +87,21 @@ pub fn execute(
     info: MessageInfo,
     msg: ExecuteMsg,
 ) -> NeutronResult<Response<NeutronMsg>> {
-    match msg {
-        ExecuteMsg::DistributeFallback { coins } => try_distribute_fallback(deps, env, info, coins),
-        ExecuteMsg::Tick {} => try_tick(deps, env, info),
+    match (msg, CONTRACT_STATE.load(deps.storage)?) {
+        // ticks coming in to ibc forwarder in Instantiaed state will attempt to register the ICA
+        (ExecuteMsg::Tick {}, ContractState::Instantiated) => {
+            verify_caller(&info.sender, &CONTRACT_OP_MODE.load(deps.storage)?)?;
+            try_register_ica(deps, env)
+        }
+        // ticks coming in to ibc forwarder in IcaCreated state will attempt to forward the funds
+        (ExecuteMsg::Tick {}, ContractState::IcaCreated) => {
+            verify_caller(&info.sender, &CONTRACT_OP_MODE.load(deps.storage)?)?;
+            try_forward_funds(env, deps)
```

Comment:
nitpick: switch parameters order, for consistency.

Reply:
that's true, state coming first is more logical! will change.